### ISSUE TITLE
Add docs for Stacsk-2.1 data structure to docs

### ISSUE THREAD

**Issue by igorsyl** [Trust-Machines/core-eng#245]
Thank you @marshallbtc for writing up [Stacks-2.1-Data-Structure](https://github.com/marshallbtc/Stacks-2.1-Data-Structure)](https://github.com/marshallbtc/Stacks-2.1-Data-Structure/blob/main/data-structure.md)!

@kenrogers I suggest, we:
- merge into the [Stacks Dev Guide](https://github.com/Trust-Machines/core-eng/blob/main/stacks-dev-guide/guide.md), and
- then move this file into https://github.com/stacks-network/docs/

**Comment by kenrogers** [comment]
Trying to avoid having generic, catch-all Q&A pages on the docs, we should see where some of this content would fit into the existing docs pages and add new ones where it makes sense to effectively capture and organize this information.

What do you think about the following?

Taking all of the existing information in the TM dev guide, integrating into the existing docs, and archiving the dev guide, adding this kind of information via PRs to the docs from now on, that way we can contain all relevant information to the docs instead of having multiple disparate sources of information.

As part of that we can add this new SQL data structure file to the docs where it makes sense as well.

**Comment by marshallbtc** [comment]
I like it! 
Some of the Description fields do need a little love before the file is added to the official documentation; for example, see the question on line 569 (in the payments table under the `/stacks-blockchain/mainnet/chainstate/vm/index.sqlite` header)

Even if most of the Description fields aren't filled out, I think the schema of each table is still helpful documentation.

**Comment by igorsyl** [comment]
Yes @kenrogers ! I think we need a section of the Stacks docs geared toward protocol developers.  I think this is what I suggested initially, sorry if my initial suggestion was confusing.  Could you add a PR to blast our stacks-dev-guide after you move it out and provide the new link to the docs? The core-eng repo is for rapid incubation of ideas and I think the dev docs are ready to hatch!
